src/reducers/ComparisonSlice.ts

```diff
@@ -23,6 +33,63 @@ const comparison = createSlice({
   },
 });
 
+function processResults(results: CompareResultsItem[]) {
+  const processedResults: Map<string, CompareResultsItem[]> = new Map<
+    string,
+    CompareResultsItem[]
+  >();
+  results.forEach((result) => {
+    const { new_rev: newRevision, header_name: header } = result;
+    const rowIdentifier = header.concat(' ', newRevision);
+    if (processedResults.has(rowIdentifier)) {
+      (processedResults.get(rowIdentifier) as CompareResultsItem[]).push(
+        result,
+      );
+    } else {
+      processedResults.set(rowIdentifier, [result]);
+    }
+  });
+  const restructuredResults: Results[] = Array.from(
+    processedResults,
+    function (entry) {
```

Comment:
nit: to make this more readable, what do you think of destructuring the parameter?
```suggestion
    function ([rowIdentifier, results]) {
```